tests/simulator_test.cpp

```diff
@@ -132,25 +122,28 @@ TEST_F(EnergyCollectorTest, NotEvenNumOfEnergyCollectorTest) {
 
   Ray alongX(kVecZero, kVecX);
   ASSERT_TRUE(performHitCollector(energyCollectors, alongX, &hitData));
-  ASSERT_EQ(collectorPositionRadius - refCollectorRadius, hitData.time);
+  ASSERT_FLOAT_EQ(collectorPositionRadius - refCollectorRadius, hitData.time);
 
   Ray alongY(kVecZero, kVecY);
   ASSERT_TRUE(performHitCollector(energyCollectors, alongY, &hitData));
-  ASSERT_EQ(collectorPositionRadius - refCollectorRadius, hitData.time);
+  ASSERT_FLOAT_EQ(collectorPositionRadius - refCollectorRadius, hitData.time);
 
-  Ray at2Angle(kVecZero, Vec3(std::cos(2 * collectorAngle), 0,
-                              std::sin(2 * collectorAngle)));
+  const float collectorAngle = 2 * kPi / (numCollectors - 1);
+  Ray at2Angle(kVecZero, Vec3(std::cos(0.5 * collectorAngle), 0,
```

Comment:
Core reuse. Make a function.

(from now on I will be just mentioning these short rules you are not following to save time).

Reply:
Done where? I do not see it.

Reply:
now done. I double checked implementation of generating ray at cerain angle and I didn't find anything wrong with that ;/